dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3985,19 +4028,31 @@ Sedp::Reader::data_received(const DCPS::ReceivedDataSample& sample)
 #endif
 
     } else if (entity_id == ENTITYID_TL_SVC_REQ_WRITER) {
-      // TODO: process request and send reply
-      XTypes::TypeLookup_Request type_lookup_request;
-      if (!sedp_.type_lookup_request_reader_->take_tl_request(sample, ser, type_lookup_request)) {
+      // TLS_TODO: verify request processing
+      XTypes::TypeLookup_Reply type_lookup_reply;
+      if (DDS::RETCODE_OK != sedp_.type_lookup_request_reader_->take_tl_request(sample, ser, type_lookup_reply)) {
         ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Sedp::Reader::data_received - ")
-          ACE_TEXT("failed to take type lookup request\n")));
+                   ACE_TEXT("failed to take type lookup request\n")));
+        return;
+      }
+
+      // TLS_TODO: verify reader
+      DCPS::RepoId reader = sample.header_.publication_id_;
+      reader.entityId = ENTITYID_TL_SVC_REPLY_READER;
+      DCPS::SequenceNumber sequence = 0;
+      if (DDS::RETCODE_OK != sedp_.type_lookup_reply_writer_->send_type_lookup_reply(type_lookup_reply,
+                                                                                     reader,
+                                                                                     sequence,
+                                                                                     DDS::RPC::REMOTE_EX_OK)) {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Sedp::Reader::data_received - ")
+                  ACE_TEXT("failed to send type lookup reply\n")));
         return;
       }
     } else if (entity_id == ENTITYID_TL_SVC_REPLY_WRITER) {
-      // TODO: process reply
-      XTypes::TypeLookup_Reply type_lookup_reply;
-      if (!sedp_.type_lookup_reply_reader_->take_tl_reply(sample, ser, type_lookup_reply)) {
+      // TLS_TODO: verify reply processing
+      if (!sedp_.type_lookup_reply_reader_->take_tl_reply(sample, ser)) {
         ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Sedp::Reader::data_received - ")
-          ACE_TEXT("failed to take type lookup reply\n")));
+                  ACE_TEXT("failed to take type lookup reply\n")));
```

Comment:
```suggestion
                   ACE_TEXT("failed to take type lookup reply\n")));
```

One more space if you're going to match the indent with the start of the arguments.